src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs

```diff
@@ -61,6 +61,18 @@ public override void Initialize(AnalysisContext context)
                 {
                     var op = (IReturnOperation)oc.Operation;
                     if (op.ReturnedValue == null) return;
+
+                    // In the abscense of data flow analysis we can treat
+                    // return of a local variable or a parameter as a "move".
+                    if ((op.ReturnedValue.Kind == OperationKind.LocalReference
+                        || op.ReturnedValue.Kind == OperationKind.ParameterReference)
+                        && op.Kind == OperationKind.Return)
+                    {
+                        var varScope = op.ReturnedValue.GetSymbol().ContainingSymbol;
+                        var opScope = op.SemanticModel.GetEnclosingSymbol(op.Syntax.SpanStart);
+                        if (SymbolEqualityComparer.Default.Equals(varScope, opScope)) return;
+                    }
+
```

Comment:
While I understand the demand, should this be included in this pull request?

Reply:
@ufcpp TBH, I was too lazy to split the changes in multiple PRs. E.g. this is not a PR to support C#, this is a PR to bring all the changes I made to enhance NonCopyableAnalyzer for our open source project https://github.com/pythonnet/pythonnet/ in one swat.

If you really insist, I can make separate PRs, but it should be easy enough to review individual commits in this one.

Reply:
OK. Don't mind. I'll merge this PR and release the package in my spare time.

Reply:
https://www.nuget.org/packages/NonCopyableAnalyzer/0.7.0